src/lib/server/hooks/handle-saved-guild.ts

```diff
@@ -0,0 +1,69 @@
+import cookie from 'cookie'
+import { Routes } from 'discord.js'
+import { GUILD_COOKIE } from '$lib/constants'
+import { api } from '$lib/discord/api'
+import { getBotGuilds } from '$lib/discord/get-bot-guilds'
+import type { Handle, RequestEvent } from '@sveltejs/kit'
+
+/**
+ * Parse cookies from the event and returns the value of the guild cookie
+ */
+function useSavedGuild(
+  event: RequestEvent<Partial<Record<string, string>>, string | null>
+) {
+  const cookies = event.request.headers.get('cookie')
+  if (cookies) {
+    const parsed = cookie.parse(cookies)
+    return parsed[GUILD_COOKIE]
+  }
+  return null
+}
+
+export const handleSavedGuild: Handle = async ({ event, resolve }) => {
+  const savedGuild = useSavedGuild(event)
+
+  /** Guilds that are shared between the current user and bot */
+  const sharedGuilds = []
+
+  // only attempt to fetch guild memberships if the user is logged in
+  if (savedGuild && event.locals.session?.user) {
+    const botGuilds = await getBotGuilds()
+    for (const guild of botGuilds) {
+      try {
+        await api.get(
+          Routes.guildMember(guild.id, event.locals.session.user.discordUserId)
+        )
```

Comment:
Could this cause an infinite loop ever since this is in middleware?

Reply:
I don't believe so. Are you thinking of a request retry causing the loop?